types/duration.go

```diff
@@ -0,0 +1,258 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+	"unicode"
+)
+
+const (
```

Comment:
Ah, since I mentioned those constants elsewhere, maybe they can all live in the `internal/consts` package?